pkg/detector/detector.go

```diff
@@ -901,12 +902,24 @@ func (d *ResourceDetector) GetMatching(resourceSelectors []policyv1alpha1.Resour
 
 // OnPropagationPolicyAdd handles object add event and push the object to queue.
 func (d *ResourceDetector) OnPropagationPolicyAdd(obj interface{}) {
-	d.policyReconcileWorker.Enqueue(obj)
+	policyObj := obj.(*policyv1alpha1.PropagationPolicy)
+	policyObj.SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
For pp/cpp, do we not need to set its gvk, because we do not need this information when consuming queues, so we can use a separate keyFunc.

Reply:
> so we can use a separate keyFunc.

In this case, do we need a new structure to replace ClusterWideKey, excluding the gvk field?

Reply:
The structure can be universal. When consuming the policy worker queue, we do not need to obtain the gvk.

Reply:
maybe newFunc just need remove gvk check logic?

https://github.com/karmada-io/karmada/blob/9ea0382293c80fdf8cf13da914136dfc4df3ec51/pkg/util/fedinformer/keys/keys.go#L96-L103

Reply:
Yes, I think so.